srml/contracts/src/exec.rs

```diff
@@ -321,14 +326,20 @@ where
 		// Assumption: pay_rent doesn't collide with overlay because
 		// pay_rent will be done on first call and dest contract and balance
 		// cannot be changed before the first call
-		crate::rent::pay_rent::<T>(&dest);
+		let contract_info = rent::pay_rent(&dest);
+
+		// Calls to dead contracts always fail.
+		if let Some(ContractInfo::Tombstone(_)) = contract_info {
```

Comment:
is it difficult to avoid it is just checking contract_into like here and return if it is none no?

But yes seems fine at the same time.

Reply:
Replied by editing original comment.